easybuild/framework/easyblock.py

```diff
@@ -2428,18 +2428,28 @@ def extensions_step(self, fetch=False, install=True):
         if self.skip:
             self.skip_extensions()
 
-        exts_cnt = len(self.ext_instances)
-        for idx, ext in enumerate(self.ext_instances):
+        # Create progress bar for extensions
+        if self.progress_bar:
+            ext_task_id = self.progress_bar.add_task("Extension", total=len(self.ext_instances), software="")
+            # Step size to advance progress bar for extensions
+            ext_step_size = 1.0 / 4.0
+
+        def _advance_ext_bar():
+            """
+            Helper method to update the extension progress bar
+            """
+            if self.progress_bar:
+                self.progress_bar.advance(ext_task_id, ext_step_size)
+
+        for ext in self.ext_instances:
 
             self.log.debug("Starting extension %s" % ext.name)
+            if self.progress_bar:
+                self.progress_bar.update(ext_task_id, software=ext.name)
 
             # always go back to original work dir to avoid running stuff from a dir that no longer exists
             change_dir(self.orig_workdir)
 
-            tup = (ext.name, ext.version or '', idx + 1, exts_cnt)
-            print_msg("installing extension %s %s (%d/%d)..." % tup, silent=self.silent)
```

Comment:
We shouldn't remove this output, since Rich is an optional dependency.

Still printing this when using Rich may even still makes sense (perhaps in a slightly different form), to give an overview of what was installed already.

The progress bar only shows what's currently being installed, it probably makes sense to keep showing what was installed already too...